### DISCUSSION TITLE
Common CDT.Cloud test project

### DISCUSSION THREAD

**Discussion by planger** [eclipse-cdt-cloud/cdt-cloud#4] [Ideas]
I propose to provide a shared test project with which all components of CDT.Cloud can be tried out by interested parties. Ideally, this should be easy to check out or create within CDT.Cloud Blueprint. A welcome page could guide through the functionality of all components based on this test project.

**Comment by MatthewKhouzam** [comment]
Any thoughts on what the project would be? I like the idea. I would like it to be able to exercise all* the sub-projects if possible.

*as many as we can reasonably.

**Reply by planger** [reply]
Good question. I think we should collect some requirements or characteristics to demonstrate the features of all components, start off with a sample project (e.g. [Pipe-Runner-Lab/sample_cmake](https://github.com/Pipe-Runner-Lab/sample_cmake)) and enhance it if necessary to meet the requirements mentioned above.

For the basic features of CDT.Cloud Blueprint, I think anything that supports the following should be fine:

- Support for generating a compilation database for clangd (e.g. CMake based projects)
- Some CPP files that allow demonstrating navigation, auto-completion, maybe macros
- Debugable with GDB

It should ideally also make use of the new [context switching support](https://github.com/eclipse-theia/theia-cpp-extensions/pull/143), once it comes in.
